### DISCUSSION TITLE
Regarding TUI

### DISCUSSION THREAD

**Discussion by simonsan** [GitoxideLabs/gitoxide#220] [Q&A] [answered]
https://github.com/Byron/gitoxide/blob/main/crate-status.md#git-tui

I wonder if it would be viable to connect to an existing project like <https://github.com/extrawurst/gitui> and replace the dependency on `libgit2/git` crate/bindings with `gitoxide`.

**Comment by Byron** [comment, marked as answer]
This can be done as part of the [oxidization effort](https://github.com/Byron/gitoxide/projects/1#card-70898500) which will soon be on the way.

Having an own go at a TUI will allow to completely rethink how we work with `git` in order to make it more approachable.

**Reply by simonsan** [reply]
> completely rethink how we work with `git` in order to make it more approachable.

I wonder what you mean by that. Lowering the barriers of entry is awesome as always, but how do you imagine it?

**Reply by Byron** [reply]
I think it's worth limiting the commands to the smallest possible subset and let them figure out smartly what to do. If there is choice, present it to the user while in interactive mode (which is the assumed mode), or abort. The TUI should be flexible, that is use the space present and optionally drop into full alternate screen mode for having more space for visualizing the commit graph. The latter must be simplified to only show what matters in the situation with options to dig in when needed. At all times it must be easy to obtain help or learn how to operate the thing, probably by also providing mouse support.

So in short, it's about implementing typical user journeys while going slightly new ways in terms of terminal user interfaces. And of course, all operations will be fast, as in as fast or faster as canonical `git`.
